src/commands/deploy/index.js

```diff
@@ -99,39 +101,51 @@ export default async ctx => {
   const file = highlight('now.json');
   const prop = code('version');
 
-  if (!localConfig) {
-    if (!isFile) {
-      output.warn(
-        `Your project is missing a ${file} file with a ${prop} property. More: https://zeit.co/docs/version-config`
-      );
+  if (!versionFlag && !localConfig && process.stdout.isTTY) {
+    if (isFile) {
+      output.error(`There was an issue parsing ${file}`);
+      return 1;
     }
-  } else {
-    const { version } = localConfig;
 
-    if (version) {
-      if (typeof version === 'number') {
-        if (version !== 1 && version !== 2) {
-          const first = code(1);
-          const second = code(2);
+    output.debug(`${file} not found, generating...`);
+    const { config } = await generateProject(paths[0], output);
+
+    if (await promptBool('Would you like to deploy?', { defaultValue: true })) {
+      localConfig = config;
+    } else {
+      return 0;
+    }
+  }
 
-          output.error(
-            `The value of the ${prop} property within ${file} can only be ${first} or ${second}.`
-          );
-          return 1;
-        }
+  const { version } = localConfig || { version: null };
```

Comment:
Did you use `{ version: null }` because you want to fallback to `null` instead of `undefined`?

Reply:
```suggestion
  const { version } = localConfig || {};
```
Would work wouldn't it?

Reply:
@dav-is  Yes that would work.

Perhaps it would be better to change the order of this if/else to make it more readable:

```js
if (!localConfig) {
  warn('missing now.json')
} else if (typeof localConfig.version === 'undefined') {
  warn('missing version key in now.json')
} else if (typeof localConfig.version !== 'number') {
  warn('the version key in now.json must be a number')
} else if (typeof localConfig.version !== 1 && localConfig.version !== 2) {
  warn('the version key in now.json can only be 1 or 2')
} else {
  platformVersion = localConfig.version;
}
```

This is kind of pseudo-code but you get the idea 😄

Reply:
You're missing the case where we generate

Reply:
Yep, that's before

Reply:
Pushed a fix :+1: